pkg/cmd/dev/build.go

```diff
@@ -110,6 +110,15 @@ func (d *dev) build(cmd *cobra.Command, commandLine []string) error {
 	args = append(args, additionalBazelArgs...)
 
 	if cross == "" {
+		// run "yarn --check-files" before any bazel target that includes UI to ensure that node_modules dir is consistent
+		// see related issue: https://github.com/cockroachdb/cockroach/issues/70867
+		for _, arg := range args {
+			if arg == "--config=with_ui" {
+				logCommand("bazel", "run", "@nodejs//:yarn", "--", "--check-files", "--cwd", "pkg/ui", "--offline")
+				d.exec.CommandContextInheritingStdStreams(ctx, "bazel", "run", "@nodejs//:yarn", "--", "--check-files", "--cwd", "pkg/ui", "--offline")
```

Comment:
This will require some updates to the test files.

Reply:
Hey @koorosh -- 

* In `pkg/cmd/dev/testdata/build.txt` you'll find a bunch of test cases along with all the OS operations each of those test cases will run. At a glance, it looks like the `dev build --skip-generate -- --verbose_failures --sandbox_debug` test case is the only one affected by your change. It looks like you will have to add a `bazel run @nodejs//:yarn -- --check-files --cwd pkg/ui --offline` line, before the build proper.
* Then in `pkg/cmd/dev/testdata/recording/build.txt` there are the recordings for all of those test cases. Add the following in the appropriate spot:

```
bazel run @nodejs//:yarn -- --check-files --cwd pkg/ui --offline
----

```

That should be it. You can DM me for any more questions.

Reply:
Oh, you can `dev test pkg/cmd/dev` to test your changes.

Reply:
Thank you @rickystewart !

Reply:
Done!